graph/telemetry/istio/appender/istio_details.go

```diff
@@ -110,30 +117,32 @@ NODES:
 }
 
 // addLabels is a chance to add any missing label info to nodes when the telemetry does not provide enough information.
-// TODO: For efficiency we may want to consider pulling all namespace service definitions in one call (the call does not
-//       exist at this writing).  As written we pull each service individually, which can be a fair number of round
-//       trips when services are injected (as they are by default). Note also that currently we do query for
-//       outsider service nodes.  That may be a security problem f the outside namespace is inaccessible to the user. If
-//       that becomes an issue we can limit to accessible namespaces or only to the AppenderNamespaceInfo namespace.
-func addLabels(trafficMap graph.TrafficMap, globalInfo *graph.AppenderGlobalInfo) {
+// For example, service injection has this problem.
+func addLabels(trafficMap graph.TrafficMap, globalInfo *graph.AppenderGlobalInfo, sdl *models.ServiceDefinitionList) {
```

Comment:
>  I understand the only impact is that services in namespaces other than the requested one won't be grouped into an app box.

That's mostly correct.  Services outside of the current namespace will not get grouped.  But if that service is in *any* of the requested namespaces it will get processed.  It is only services outside of any requested namespaces that will not be processed.  As mentioned in the original comment, that may have been a potential security issue anyway, so I don't think it's a big deal considering the perf improvement we should be getting with the new approach.